src/runtime/database.cpp

```diff
@@ -1210,6 +1220,36 @@ static std::vector<JobReflection> find_all(const Database *db, sqlite3_stmt *que
   return out;
 }
 
+static std::vector<JobOutput> failed_job_outputs(const Database *db, sqlite3_stmt *query) {
```

Comment:
With the change to use JobReflection you can delete this code but you'll need to update `find_one` and `db->get_output` to use the new interwoven style for stdout and stderr